### PR TITLE
Implement ad set selector depending on current ad account

### PR DESCRIPTION
**Pull request by yasu**
I added an ad set selector to be able to get ads insights only from a specified ad set. This helps requesting big amount of data from Facebook to cause following error.
> Please reduce the amount of data you're asking for, then retry your request

@axross Could you please review my code?

### CODE REVIEW

File: src/js/actions/index.js [hunk 1]
```diff
@@ -1,28 +1,57 @@
...
-export const completeFetchAdaccounts = () => ({ type: COMPLETE_FETCH_ADACCOUNTS });
-export const fetchAdaccounts = () => (dispatch) => {
-  dispatch(requestFetchAdaccounts());
-  fb.api('v2.8/me/adaccounts', { fields: 'id,name' }, (response) => {
+export const requestFetchAdSets = () => ({ type: REQUEST_FETCH_ADSETS });
+export const completeFetchAdSets = () => ({ type: COMPLETE_FETCH_ADSETS });
+export const fetchAdSets = (accountId) => {
+  return (dispatch) => {
+    dispatch(requestFetchAdSets());
+    fb.api(`v2.8/${accountId}/adsets`, { fields: 'id,name', limit: 100 }, (response) => {
```
Comment (axross) [inline src/js/actions/index.js]: I think parenthesis `()` are not necessary in this line and above line. It's okay if you want to do it.

File: src/js/actions/index.js [hunk 2]
```diff
@@ -1,28 +1,57 @@
...
+export const selectAdSet = id => ({
+  type: SELECT_ADSET,
+  payload: { id },
+});
+
+export const requestFetchAdAccounts = () => ({ type: REQUEST_FETCH_ADACCOUNTS });
+export const completeFetchAdAccounts = () => ({ type: COMPLETE_FETCH_ADACCOUNTS });
+export const fetchAdAccounts = () => (dispatch) => {
+  dispatch(requestFetchAdAccounts());
+  fb.api('v2.8/me/adaccounts', { fields: 'id,name', limit: 100 }, (response) => {
```
Comment (axross) [inline src/js/actions/index.js]: Same as the above comment.

File: src/js/reducers/adAccounts.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+const initialState = { all: {}, current: null, isFetching: false };
+
+const adAccounts = (state = initialState, action) => {
+  switch (action.type) {
+    case REQUEST_FETCH_ADACCOUNTS:
+      return { ...state, all: {}, current: null, isFetching: true };
+    case COMPLETE_FETCH_ADACCOUNTS:
+      return { ...state, isFetching: false };
+    case FETCH_ADACCOUNTS:
+      return { ...state, all: action.payload.data.reduce((h, x) => Object.assign(h, { [x.id]: x }), {}) };
```
Comment (axross) [inline src/js/reducers/adAccounts.js]: I recommend you use instead Object.assign({}, foo, bar ...) as immutable. Array#reduce() returns a created new array. But this code mutate the original array.

File: src/js/reducers/adSets.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+const initialState = { all: {}, current: null, isFetching: false };
+
+const adSets = (state = initialState, action) => {
+  switch (action.type) {
+    case REQUEST_FETCH_ADSETS:
+      return { ...state, all: {}, current: null, isFetching: true };
+    case COMPLETE_FETCH_ADSETS:
+      return { ...state, isFetching: false };
+    case FETCH_ADSETS:
+      return { ...state, all: action.payload.data.reduce((h, x) => Object.assign(h, { [x.id]: x }), {}) };
```
Comment (axross) [inline src/js/reducers/adSets.js]: Same as the above comment.

File: src/js/actions/index.js (hunk 1 above)
Comment (axross) [inline src/js/actions/index.js:18]: I think parenthesis () are not necessary in this line and above line. It's okay if you want to do it.

File: src/js/actions/index.js (hunk 2 above)
Comment (axross) [inline src/js/actions/index.js:37]: Same as the above comment.

**Review by axross — COMMENTED** [review]
I commented but it is LGTM.

**Review by axross — APPROVED** [review]
I approve to merge.

**Comment by yasu** [comment]
@axross Thanks for your review! I will leave your points because the default settings of `gulp-eslint` requires the parentheses ().

**Pull request merged**